nautobot/utilities/utils.py

```diff
@@ -672,3 +695,164 @@ def pretty_str(self, node=None, depth=0):
+        {
+            "id": name,
+            "name": build_lookup_label(name, field.lookup_expr),
+        }
+        for name, field in filterset.items()
+        if name.startswith(field_name) and not name.startswith("has_")
+    ]
+    return lookup_expr or [{"id": "exact", "name": "exact - Not Found"}]
+
+
+def get_data_for_filterset_parameter(model, parameter, initial_choice=None):
+    """
+    Return relevant data for a filterset parameter which can include `model` api_url, the parameter-type,
+    parameter choices etc.
+
+    Args:
+        model: Filterset model
+        parameter: The filterset parameter e.g has_vlans, last_updated__lte, slug e.t.c
+        initial_choice: [Optional] The filterset parameter value
+
+    Examples:
+        >>> get_data_for_filterset_parameter(<class: nautobot.dcim.models.Site>, "has_vlans")
+        >>> {"type": "static-choice", "choices": [["True", "Yes", ["False", "No"]]], "allow_multiple": False}
+
+        >>> get_data_for_filterset_parameter(<class: nautobot.dcim.models.Site>, "status")
+        >>> {"type": "dynamic-choice", "choices": [], "data_url": "/api/dcim/sites/"}
+
+    Returns:
+        Returns a dict which keys may vary depending on the parameter type
+        e.g relational filed(ForeignKey, ManyToMany e.t.c.), choice field including boolean or char/int field:
+            type: param type either staic-choice/dynamic-choice/others,
+            data_url[for relational field]: param models api endpoint,
+            content_type[for relational field]: (model contenttype),
+            choices[for choice field or if initial_choice is not None]:  param available choices,
+            allow_multiple[for choice field]:  allow multiple choices or not
+    """
+    # Avoid circular import
+    from nautobot.extras.filters import StatusFilter
+    from nautobot.extras.models import Status, Tag
+    from nautobot.utilities.filters import RelatedMembershipBooleanFilter
+
+    contenttype = model._meta.app_label + "." + model._meta.model_name
+
+    filterset = get_filterset_for_model(model)
+    if filterset is None:
+        # TODO Timizuo Raise Error if filterset not found
+        pass
+
+    field = filterset.base_filters.get(parameter)
+    if field is None:
+        # TODO Timizuo Raise Error if field not found
+        pass
+
+    data = {"type": "others"}
+
+    if isinstance(field, (filters.MultipleChoiceFilter, ModelMultipleChoiceFilter)):
+        if "choices" in field.extra:  # Field choices
+            # Problem might arise here if plugin developer do not declare field inheriting from nautobot.utilities.choices.ChoiceSet
+            data = {
+                "type": "static-choices",
+                "choices": field.extra["choices"].CHOICES,
+                "allow_multiple": True,
+            }
+        elif hasattr(field, "queryset"):  # Dynamically populated choices
+            if isinstance(field, StatusFilter):
+                related_model = Status
```

Comment:
Why do we need this special case?

Reply:
`StatusFilter` do not have the attribute `queryset`.

Reply:
Should it, perhaps?